src/Illuminate/Foundation/Http/FormRequest.php

```diff
@@ -103,6 +103,16 @@ protected function createDefaultValidator(ValidationFactory $factory)
     }
 
     /**
+     * Get the validated data from the request.
+     *
+     * @return array
+     */
+    public function validated()
+    {
+        return $this->only(array_keys($this->container->call([$this, 'rules'])));
```

Comment:
@rodrigopedra `$this->container->call([$this, 'rules'])` calls the method and inject any dependencies it has declared, as method parantheses. `$this->rules()` doesn't.